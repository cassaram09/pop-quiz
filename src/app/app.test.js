import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, render, mount } from 'utils/enzyme';

import App from './app';

describe('App', () => {
  it('renders without crashing', () => {
    const app = shallow(<App />);
  
    expect(app).toMatchSnapshot();
  });

  it('has the correct text', () => {
    const app = render(<App />);
    const p = app.find('p'); 
    
    expect(p.text()).toBe('Edit src/App.js and save to reload.'); // fails,
  });
});
