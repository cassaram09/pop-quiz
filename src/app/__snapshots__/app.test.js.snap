// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Edit ",
          <code>
            src/App.js
          </code>,
          " and save to reload.",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Edit ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "src/App.js",
          },
          "ref": null,
          "rendered": "src/App.js",
          "type": "code",
        },
        " and save to reload.",
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Edit 
          <code>
            src/App.js
          </code>
           and save to reload.
        </p>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Edit ",
            <code>
              src/App.js
            </code>,
            " and save to reload.",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Edit ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "src/App.js",
            },
            "ref": null,
            "rendered": "src/App.js",
            "type": "code",
          },
          " and save to reload.",
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
